lib/Container.js

```diff
@@ -63,12 +58,38 @@ Container.prototype.buildSetupTasks = Promise.promisify(function(done) {
   done(null, tasks);
 });
 
+Container.prototype._createDownloader = function(build, project){
+  // map provider type to downloader class name
+  var downloaders = {
+    'github': 'GithubDownloader',
+    'stash': 'StashDownloader',
+  }
+
+  var provider_type = project.provider.type
+  var clazz = downloaders[provider_type]
+  if(!clazz){
+    throw new Error("Download: unsupported provider type: " + provider_type)
+  }
+
+  // dynamically require and instantiate a downloder
+  var Downloader
+  try {
+    Downloader = require(`./plugins/TaskRunner/${clazz}`)
+  } catch (e) {
+    var msg = `Download: failed to load supported downloader for provider ${provider_type}: ${e.message}`
+    this.log.error({err: e, clazz}, msg)
```

Comment:
We shouldn't lean implementation details like the `clazz` variable name into the error log.